exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -2,168 +2,148 @@
+
     LinkedList popQuestions = new LinkedList();
     LinkedList scienceQuestions = new LinkedList();
     LinkedList sportsQuestions = new LinkedList();
     LinkedList rockQuestions = new LinkedList();
-    
+    List<String> board = new ArrayList<>();
+    static final int NUMBER_OF_QUESTIONS = 50;
     int currentPlayer = 0;
     boolean isGettingOutOfPenaltyBox;
-    
-    public  Game(){
-    	for (int i = 0; i < 50; i++) {
-			popQuestions.addLast("Pop Question " + i);
-			scienceQuestions.addLast(("Science Question " + i));
-			sportsQuestions.addLast(("Sports Question " + i));
-			rockQuestions.addLast(createRockQuestion(i));
-    	}
+
+    public Game() {
+        for (int i = 0; i < NUMBER_OF_QUESTIONS; i++) {
+            popQuestions.addLast(createQuestion(i, "Pop"));
+            scienceQuestions.addLast(createQuestion(i, "Science"));
+            sportsQuestions.addLast(createQuestion(i, "Sports"));
+            rockQuestions.addLast(createQuestion(i, "Rock"));
+        }
+        for (int i = 0; i < 3; i++) {
+            board.add("Pop");
+            board.add("Science");
+            board.add("Sports");
+            board.add("Rock");
+        }
+    }
+
+    public String createQuestion(int index, String category) {
+        return (category + " Question " + index);
+    }
+
+    public void addNewPlayer(String playerName) {
+        players.add(playerName);
+        places[numberOfPlayers()] = 0;
+        coinCount[numberOfPlayers()] = 0;
+        isInPenaltyBox[numberOfPlayers()] = false;
+
+        System.out.println(playerName + " was added");
+        System.out.println("They are player number " + players.size());
+
+    }
+
+    public int numberOfPlayers() {
+        return players.size();
+    }
+
+    public void turn(int roll) {
+        System.out.println(players.get(currentPlayer) + " is the current player");
+        System.out.println("They have rolled a " + roll);
+        if (isInPenaltyBox[currentPlayer]) {
+            if (roll % 2 != 0) {
+                isGettingOutOfPenaltyBox = true;
+                System.out.println(players.get(currentPlayer) + " is getting out of the penalty box");
+                movePlayer(roll);
+                askQuestion();
+            } else {
+                System.out.println(players.get(currentPlayer) + " is not getting out of the penalty box");
+                isGettingOutOfPenaltyBox = false;
+            }
+
+        } else {
+            movePlayer(roll);
+            askQuestion();
+        }
+
+    }
+
+    private void askQuestion() {
+        if (currentCategory() == "Pop")
+            System.out.println(popQuestions.removeFirst());
+        if (currentCategory() == "Science")
+            System.out.println(scienceQuestions.removeFirst());
+        if (currentCategory() == "Sports")
+            System.out.println(sportsQuestions.removeFirst());
+        if (currentCategory().equals("Rock"))
+            System.out.println(rockQuestions.removeFirst());

```

Comment:
Replacing these conditionals with a class that handles the categories may reduce complexity here.
A board class in itself, which also handles the player's movements, could cover this.

Reply:
A great suggestion, thank you